enterprise/tpa_pipeline.py

```diff
@@ -79,6 +79,10 @@ def handle_enterprise_logistration(backend, user, **kwargs):
             handle_redirect_after_social_auth_login(backend, user)
         return
 
+    if not enterprise_customer.identity_providers.first().identity_provider.is_valid:
```

Comment:
any chance `enterprise_customer.identity_providers` could be empty and thus `.first()` would blow up? Its always the first provider?

Also, we renamed this field `was_valid_at`

Reply:
We may be adding axsecond sso config for example so I think we won’t want to assume first instead if possiblecwe want to use the exact  config we are working with

Reply:
I believe that this `handle_enterprise_logistration` method is only hit when we have an SSO interaction, so the existence of an identity provider should be guaranteed

Reply:
agreed with that, but what I meant is what if there are multiple items in identity_providers? should we not update the correct one? I have not checked, but is it possible to know which provider config we are working with, in this part of the flow ? I can try to look that up

Reply:
I think you just want to use that provider config, based on this id, rather than first()

Reply:
I've changed the code to filter for the provider id 👍